automap/src/comm_layer/pcp_pmp_common/linux_specific.rs

```diff
@@ -190,42 +192,80 @@ mod tests {
     }
 
     #[test]
-    fn reg_for_ip_route_command_output() {
-        let route_n_output = "\
-        10.1.0.0/16 dev eth0 proto kernel scope link src 10.1.0.84 metric 100\n\
-        0.1.0.1 dev eth0 proto dhcp scope link src 10.1.0.84 metric 100\n\
-        168.63.129.16 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100\n\
-        169.254.169.254 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100\n\
-        172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown";
-        let regex = ip_route_regex();
-
-        let lines = route_n_output.split('\n').collect::<Vec<&str>>();
-
-        lines.iter().for_each(|line| {
-            assert!(regex.is_match(line), "Lines: {:?} line: {}", lines, line);
-        });
-    }
-
-    #[test]
-    fn reg_for_ip_route_bad_ip() {
-        let route_n_output = "\
-        10.1.0.0/16 dev eth0 proto kernel scope link src 10.1.0.84 metric 100\n\
-        0.1.0 dev eth0 proto dhcp scope link src 10.1.0.84 metric 100\n\
-        0.1.255.1 dev eth0 proto dhcp\n\
-        0.1.256.1 dev eth0 proto dhcp\n\
-        0.1.b.1 dev eth0 proto dhcp\n\
-        0.1.0.1/ dev eth0 proto dhcp\n\
-        2001:0db8:0000:0000:0000:ff00:0042:8329 dev eth0 proto dhcp";
+    fn reg_for_ip_route_command_output_good_and_bad_ip() {
+        let route_n_output = vec![
+            (
+                "default via 0.1.0.1 dev eth0 proto dhcp scope link src 10.1.0.84 metric 100",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "10.1.0.0/16 dev eth0 proto kernel scope link src 10.1.0.84 metric 100",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "168.63.129.16 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "169.254.169.254 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "10.1.0.0/16 dev eth0 proto kernel scope link src 10.1.0.84 metric 100",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "0.1.255.1 dev eth0 proto dhcp",
+                true,
+                "Example of good IPv4",
+            ),
+            (
+                "0.1.0 dev eth0 proto dhcp scope link src 10.1.0.84 metric 100",
+                false,
+                "IPv4 address has only three elements",
+            ),
+            (
+                "0.1.256.1 dev eth0 proto dhcp",
+                false,
+                "IPv4 address is malformed",
```

Comment:
Perhaps "IPv4 address has an element greater than 255"--but it won't prevent an approval